configure.ac

```diff
@@ -748,6 +751,28 @@ AC_ARG_ENABLE(
   [AS_HELP_STRING([--disable-doxygen-version],
   [Substitute the Doxygen version with latest, for the website])])
 
+# systemd support
+AC_ARG_ENABLE(
+  [systemd],
+  [AS_HELP_STRING([--disable-systemd], [Disable systemd support])])
+
+have_libsystemd="no"
+AS_IF([test "x$enable_systemd" != "xno"],
+  [PKG_CHECK_MODULES([libsystemd], [libsystemd >= 38], [have_libsystemd="yes"], 
+                     [AC_MSG_WARN([disabling systemd support: can't find libsystemd])])
+   AC_CHECK_HEADER([systemd/sd-daemon.h], [],
+                   [have_libsystemd="no"
+                    AC_MSG_WARN([disabling systemd support: can't find systemd/sd-daemon.h])])
+   AS_IF([test "x$ac_cv_func_strerror_r" = "xno"],
+         [have_libsystemd="no"
+          AC_MSG_WARN([disabling systemd support: can't find strerror_r])])])
```

Comment:
The WARNs mean configure will continue. It might be nice to have a final bit of logic where if `"x$enable_systemd" != "xyes"` and `"x$have_libsystemd"=="xno"` it throws an error. So if you've explicitly added --enable-systemd to your configure, it bombs out, but if you've done nothing, it just warns and carries on.